photon-lib/src/main/java/org/photonvision/SimPhotonCamera.java

```diff
@@ -18,53 +18,125 @@
-    *
-    * @param cameraName The nickname of the camera (found in the PhotonVision UI).
-    */
+     * Constructs a Simulated PhotonCamera from the name of the camera.
+     *
+     * @param cameraName The nickname of the camera (found in the PhotonVision UI).
+     */
     public SimPhotonCamera(String cameraName) {
-        super(cameraName);
+        this(NetworkTableInstance.getDefault().getTable("photonvision").getSubTable(cameraName));
     }
 
     /**
-    * Simulate one processed frame of vision data, putting one result to NT.
-    *
-    * @param latencyMillis
-    * @param targets Each target detected
-    */
+     * Simulate one processed frame of vision data, putting one result to NT.
+     *
+     * @param latencyMillis Latency of the provided frame
+     * @param targets       Each target detected
+     */
     public void submitProcessedFrame(double latencyMillis, PhotonTrackedTarget... targets) {
         submitProcessedFrame(latencyMillis, Arrays.asList(targets));
     }
 
     /**
-    * Simulate one processed frame of vision data, putting one result to NT.
-    *
-    * @param latencyMillis
-    * @param tgtList List of targets detected
-    */
-    public void submitProcessedFrame(double latencyMillis, List<PhotonTrackedTarget> tgtList) {
-        if (!getDriverMode()) {
-            PhotonPipelineResult newResult = new PhotonPipelineResult(latencyMillis, tgtList);
-            var newPacket = new Packet(newResult.getPacketSize());
-            newResult.populatePacket(newPacket);
-            rawBytesEntry.setRaw(newPacket.getData());
+     * Simulate one processed frame of vision data, putting one result to NT.
+     *
+     * @param latencyMillis Latency of the provided frame
+     * @param sortMode      Order in which to sort targets
+     * @param targets       Each target detected
+     */
+    public void submitProcessedFrame(
+            double latencyMillis, PhotonTargetSortMode sortMode, PhotonTrackedTarget... targets) {
+        submitProcessedFrame(latencyMillis, sortMode, Arrays.asList(targets));
+    }
+
+    /**
+     * Simulate one processed frame of vision data, putting one result to NT.
+     *
+     * @param latencyMillis Latency of the provided frame
+     * @param targetList    List of targets detected
+     */
+    public void submitProcessedFrame(double latencyMillis, List<PhotonTrackedTarget> targetList) {
+        submitProcessedFrame(latencyMillis, null, targetList);
+    }
+
+    /**
+     * Simulate one processed frame of vision data, putting one result to NT.
+     *
+     * @param latencyMillis Latency of the provided frame
+     * @param sortMode      Order in which to sort targets
+     * @param targetList    List of targets detected
+     */
+    public void submitProcessedFrame(
+            double latencyMillis, PhotonTargetSortMode sortMode, List<PhotonTrackedTarget> targetList) {
+        latencyMillisEntry.setDouble(latencyMillis);
+
```

Comment:
One possibility here - add an if statement to catch driver mode active or LED's off - in both cases, reset target list to be empty.

The interpretation here is that targetList represents things that are in the camera's sensor, but that if statement controls whether conditions are right for photon to actually convert to processed pieces of target data.

Alternately, one layer up (in SimVisionSystem), add that similar logic to not bother running any calcualtion and empty the frame if we're in driver mode or LED's off.

Reply:
Assuming no targets with LEDs off means something like power cell tracking may fail. Drivermode = no targets is however a safe assumption.
Not sure about clearing the target list. That is not the behavior I'd expect as a user